src/main/java/kitchenpos/application/MenuService.java

```diff
@@ -1,84 +1,61 @@
 package kitchenpos.application;
 
-import kitchenpos.dao.MenuDao;
-import kitchenpos.dao.MenuGroupDao;
-import kitchenpos.dao.MenuProductDao;
-import kitchenpos.dao.ProductDao;
-import kitchenpos.domain.Menu;
-import kitchenpos.domain.MenuProduct;
-import kitchenpos.domain.Product;
+import kitchenpos.domain.*;
+import kitchenpos.dto.MenuProductRequest;
+import kitchenpos.dto.MenuRequest;
+import kitchenpos.dto.MenuResponse;
+import kitchenpos.port.MenuGroupPort;
+import kitchenpos.port.MenuPort;
+import kitchenpos.port.ProductPort;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
-import java.math.BigDecimal;
-import java.util.ArrayList;
 import java.util.List;
-import java.util.Objects;
+import java.util.stream.Collectors;
 
 @Service
+@Transactional
 public class MenuService {
-    private final MenuDao menuDao;
-    private final MenuGroupDao menuGroupDao;
-    private final MenuProductDao menuProductDao;
-    private final ProductDao productDao;
+    private final MenuPort menuPort;
+    private final MenuGroupPort menuGroupPort;
+    private final ProductPort productPort;
 
     public MenuService(
-            final MenuDao menuDao,
-            final MenuGroupDao menuGroupDao,
-            final MenuProductDao menuProductDao,
-            final ProductDao productDao
+            final MenuPort menuPort,
+            final MenuGroupPort menuGroupPort,
+            final ProductPort productPort
     ) {
-        this.menuDao = menuDao;
-        this.menuGroupDao = menuGroupDao;
-        this.menuProductDao = menuProductDao;
-        this.productDao = productDao;
+        this.menuPort = menuPort;
+        this.menuGroupPort = menuGroupPort;
+        this.productPort = productPort;
     }
 
-    @Transactional
-    public Menu create(final Menu menu) {
-        final BigDecimal price = menu.getPrice();
+    public MenuResponse create(final MenuRequest request) {
+        MenuGroup menuGroup = menuGroupPort.findById(request.getMenuGroupId());
+        List<Product> product = productPort.findAllByIdIn(getProductId(request));
```

Comment:
```suggestion
        List<Product> products = productPort.findAllByIdIn(getProductId(request));
```
복수형 변수명을 사용하는게 좋을듯하네요 🤔